cmd/lakectl/cmd/fs.go

```diff
@@ -115,28 +115,28 @@ var fsCatCmd = &cobra.Command{
 	Short: "Dump content of object to stdout",
 	Args:  cobra.ExactArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
-		client := getClient()
 		pathURI := MustParsePathURI("path", args[0])
 		direct, _ := cmd.Flags().GetBool("direct")
-		var contents []byte
+		var err error
+		var body io.ReadCloser
+		var resp *http.Response
+		client := getClient()
 		if direct {
-			_, body, err := helpers.ClientDownload(cmd.Context(), client, pathURI.Repository, pathURI.Ref, *pathURI.Path)
-			if err != nil {
-				DieErr(err)
-			}
-			defer body.Close()
-			contents, err = io.ReadAll(body)
-			if err != nil {
-				DieErr(err)
-			}
+			_, body, err = helpers.ClientDownload(cmd.Context(), client, pathURI.Repository, pathURI.Ref, *pathURI.Path)
 		} else {
-			resp, err := client.GetObjectWithResponse(cmd.Context(), pathURI.Repository, pathURI.Ref, &api.GetObjectParams{
+			resp, err = client.GetObject(cmd.Context(), pathURI.Repository, pathURI.Ref, &api.GetObjectParams{
```

Comment:
My reading of the generated code for `GetObjectWithResponse` is that `GetObject` **does _not_** handle HTTP status codes in any way.  Assuming this is true, if I `GetObject` with a bad path (say one that does not exist) -- this code will not notice the 404 status code and claim that it succeeded.

Reply:
Good point. The workflow tests in Github did identify this as well. Fixed

Reply:
Previously this was less important -- we just used the existing known-good method.  Now that we've seen it happen, can we add a test?  (That can happen in a future PR -- just open an issue, and do it after you merge this one.)

Reply:
@shimi9276 please note: PR is good.  The (only) thing blocking my approval is adding a test that _will_ identify this error.  Now that  #1657 has been merged, please add a test to CLI testing.

If you think it would take more than 1-2 hours, please instead open an issue and reply to this comment referring to that issue.

Thanks!